orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/Ec2Resource.scala

```diff
@@ -56,39 +56,64 @@ case class Ec2Resource(name: String, spec: Ec2Resource.Spec) extends ResourceIO
+      .toSet
+      .contains((instSpec \ "ec2InstanceId").as[String])
+    ssmClient.close()
+    isVisible
+  }
+
   override def getStatus(instSpec: JsValue): Either[Throwable, Status.Value] = {
     logger.debug(s"getStatus: instSpec=${instSpec.toString()}")
     val client = Client.ec2()
     val ec2InstanceId = (instSpec \ "ec2InstanceId").as[String]
-    val response: DescribeInstancesResponse = client.describeInstances(
-      DescribeInstancesRequest
-        .builder()
-        .instanceIds(ec2InstanceId)
-        .build()
-    )
-    val state = response
-      .reservations()
-      .asScala
-      .flatMap { r => r.instances().asScala.map { i => i.state().name() } }
-      .toList
-    client.close()
-    val res = state.nonEmpty match {
-      case true =>
-        state.head match {
-          case InstanceStateName.PENDING => Right(Status.Activating)
-          case InstanceStateName.RUNNING => Right(Status.Running)
-          case InstanceStateName.TERMINATED => Right(Status.Finished)
-          case InstanceStateName.STOPPING => Right(Status.Finished)
-          case InstanceStateName.STOPPED => Right(Status.Finished)
-          case InstanceStateName.SHUTTING_DOWN => Right(Status.Finished)
-          case InstanceStateName.UNKNOWN_TO_SDK_VERSION =>
-            Left(new Exception("UNKNOWN_TO_SDK_VERSION"))
-          case _ => Left(new Exception("UNKNOWN_TO_SDK_VERSION"))
-        }
-      case _ => Left(new Exception("UNKNOWN_TO_SDK_VERSION"))
+    try {
+      val response: DescribeInstancesResponse = client.describeInstances(
+        DescribeInstancesRequest
+          .builder()
+          .instanceIds(ec2InstanceId)
+          .build()
+      )
+      val state = response
+        .reservations()
+        .asScala
+        .flatMap { r => r.instances().asScala.map { i => i.state().name() } }
+        .toList
+      client.close()
+      val res = state.nonEmpty match {
+        case true =>
+          state.head match {
+            case InstanceStateName.PENDING => Right(Status.Activating)
+            case InstanceStateName.RUNNING =>
+              isSsmVisible(Json.toJson(Ec2Resource.InstSpec(ec2InstanceId))) match {
+                case true => Right(Status.Running)
+                case _ => Right(Status.Activating)
+              }
+            case InstanceStateName.TERMINATED => Right(Status.Finished)
+            case InstanceStateName.STOPPING => Right(Status.Finished)
+            case InstanceStateName.STOPPED => Right(Status.Finished)
+            case InstanceStateName.SHUTTING_DOWN => Right(Status.Finished)
+            case InstanceStateName.UNKNOWN_TO_SDK_VERSION =>
+              Left(new Exception("UNKNOWN_TO_SDK_VERSION"))
+            case _ => Left(new Exception("UNKNOWN_TO_SDK_VERSION"))
+          }
+        case _ => Left(new Exception("UNKNOWN_TO_SDK_VERSION"))
+      }
+      logger.debug(s"getStatus: result = $res")
+      res
+    } catch {
+      case e: Throwable => Left(e)
```

Comment:
what potential exceptions are we expecting here?

Reply:
when testing locally just on the Ec2Resource, I created from command line an ec2 instance, and passed in a wrong instanceId (due to typo), and got an Ec2Exception from ec2client.describeInstances() call

software.amazon.awssdk.services.ec2.model.Ec2Exception: Invalid id: "i-04efef9282d2ee064kkk" (Service: Ec2, Status Code: 400, Request ID: 69df7559-351e-43e5-9abc-b8eea71fb742)

Normally though, the Ec2Resource class would not be called with getStatus having a wrong instanceId value.  I think even the terminated instances would be around for a while with status = terminated.